lib/cloudinary/utils.rb

```diff
@@ -677,6 +678,49 @@ def self.sign_request(params, options={})
     params
   end
 
+  # Helper method for generating download URLs
+  #
+  # @param [String] action @see Cloudinary::Utils.cloudinary_api_url
+  # @param [Hash] params Query parameters in generated URL
+  # @param [Hash] options Additional options
+  # @yield [query_parameters] Invokes the block with query parameters to override how to encode them
+  #
+  # @return [String]
+  def self.cloudinary_api_download_url(action, params, options = {})
+    cloudinary_params = sign_request(params.merge(mode: MODE_DOWNLOAD), options)
+
+    cloudinary_params = if block_given?
```

Comment:
I don't see any reason why we would want to use custom function at this point. I would prefer to use Rails built-in function when it's available. Regarding tests, you can url-decode the resulting URLs and compare them against expected results.